src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -364,9 +364,13 @@ private void LogReturnedToken(AuthenticationResult result)
                 int appHashCode = AuthenticationRequestParameters.AppConfig.GetHashCode();
                 string scopes = string.Join(" ", result.Scopes);
                 Uri canonicalAuthority = new Uri(AuthenticationRequestParameters.AuthorityInfo.CanonicalAuthority);
-                AuthenticationRequestParameters.RequestContext.Logger.InfoPii(
-                    $"Fetched access token from host {canonicalAuthority.Host}. Endpoint {canonicalAuthority}. ",
-                    $"Fetched access token from host {canonicalAuthority.Host}. ");
+
+                if (result.AuthenticationResultMetadata.TokenSource != TokenSource.Cache)
+                {
+                    AuthenticationRequestParameters.RequestContext.Logger.InfoPii(
+                        $"Fetched access token from host {canonicalAuthority.Host}. Endpoint {canonicalAuthority}. ",
+                        $"Fetched access token from host {canonicalAuthority.Host}. ");
+                }
 
                 AuthenticationRequestParameters.RequestContext.Logger.Info("\n\t=== Token Acquisition finished successfully:");
                 AuthenticationRequestParameters.RequestContext.Logger.InfoPii(
```

Comment:
Log here has the same issue. Maybe just remove token source and authority altogether, we already log this.